src/core/qgsfileutils.cpp

```diff
@@ -228,3 +244,41 @@ QStringList QgsFileUtils::findFile( const QString &file, const QString &basePath
   return foundFiles;
 }
 
+
+
+QgsFileSearchTask::QgsFileSearchTask( const QString &file, const QString &basePath, int maxClimbs, int searchCeilling, const QString &currentDir, QProgressDialog *dialog, QStringList *container  )
+  : QgsTask( tr( "searching for %1" ).arg( file ), QgsTask::CanCancel )
+  , mFile( file )
+  , mBasePath( basePath )
+  , mMaxClimbs( maxClimbs )
+  , mSearchCeil( searchCeilling )
+  , mCurrentDir( currentDir )
+  , mDialog( dialog )
+  , mResults( container )
+{
+  connect( dialog, QProgessDialog::canceled, this, QgsFileSearchTask::cancel );
+}
+
+void QgsFileSearchTask::cancel()
```

Comment:
You can drop all the custom cancellation code here -- the base class versions are sufficient

Reply:
Done